.editorconfig

```diff
@@ -0,0 +1,29 @@
+#
+# MIT License
+#
+# Copyright (c) 2010 - 2020 The OSHI Project Contributors: https://github.com/oshi/oshi/graphs/contributors
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
```

Comment:
Don't commit your IDE settings files like this.  There should be an option to "not track" these.  If you'd like to add `.editorconfig` to the `.gitignore` file that's fine.

Reply:
This is not really IDE specific configuration. You can use .editorconfig files to configure almost all IDEs: https://editorconfig.org/

You can match the config to your code style guidelines. So it's easier for contributors to comply to your code style - it's rather difficult to setup your IDE for multiple different projects.

I can update the editorconfig with some default / recommended values or remove it entirely, if you want.

Reply:
I'm ok with it as long as the settings are clearly documented with comments explaining them.

Reply:
After reading the website, I'm actually rather fond of this... also add 4-space indent, and if you want to peruse the eclipse settings in the `oshi-dist` project under the resources and copy equivalenet ones, that'd be helpful.  I note that file is compatible with Eclipse (with plugin) but I don't see it referencing IntelliJ IDEA?

Reply:
IntelliJ doesn't need a plugin since quite a few versions. In addition to the standard editorconfig settings, IntelliJ supports hundreds of own configuration keys starting with `ij_`, like the two you can already see in the pull request.

I will add a few default settings then!

Reply:
I installed the Eclipse plugin. I don't see a way to export my existing eclipse file.

Spitballing without any experience, this is what I'd guess is needed. Feel free to add any  you think are missing.
```
# top-most EditorConfig file
root = true

# Unix-style newlines with a newline ending every file
[*]
end_of_line = lf
insert_final_newline = true

# Matches multiple files with brace expansion notation
# Set default charset
[*.{java,xml}]
charset = utf-8

# 4 space indentation
[*.{java,xml}]
indent_style = space
indent_size = 4

# Matches the exact files either package.json or .travis.yml
[{package.json,.travis.yml}]
indent_style = space
indent_size = 2
```

Reply:
I pushed a suggestion for the editorconfig.